spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -339,13 +341,14 @@ public void doManualAck(LinkedList<MessageHeaders> messageHeaders) {
 
 		private static ConcurrentMap<String, MimeType> mimeTypesCache = new ConcurrentHashMap<>();
 
-		static MimeType mimeTypeFromObject(Object payload) {
+		static MimeType mimeTypeFromObject(Object payload, String originalContentType) {
 			Assert.notNull(payload, "payload object cannot be null.");
 			if (payload instanceof byte[]) {
 				return MimeTypeUtils.APPLICATION_OCTET_STREAM;
 			}
 			if (payload instanceof String) {
-				return MimeTypeUtils.TEXT_PLAIN;
+				return MimeTypeUtils.APPLICATION_JSON_VALUE.equals(originalContentType) ? MimeTypeUtils.APPLICATION_JSON
+						: MimeTypeUtils.TEXT_PLAIN;
```

Comment:
Yeah, I agree. We don't have an OOTB converter for XML though, and JSON is fairly popular . One of the changes intended for 1.1 would be https://github.com/spring-cloud/spring-cloud-stream/issues/505 and we can explore adressing this in a more systematic fashion.